pipeline_steps/artifact_build.groovy

```diff
@@ -108,4 +108,30 @@ def container(String image) {
   } // pubcloud slave
 }
 
+def cloudimage(Map args) {
+  inventory="inventory.${common.rand_int_str()}"
+  inventory_path="${WORKSPACE}/rpc-gating/playbooks/${inventory}"
+  String instance_name = common.gen_instance_name()
+  try {
+    pubcloud.getPubCloudSlave(
+      instance_name: instance_name,
+      image: args.src_image,
+      region: args.region
+    )
+    pubcloud.savePubCloudSlave(
```

Comment:
I like this high level function, It doesn't seem like much of an extension to do getpubcloudslave, aio deploy, savepubcloudslave. Though obviously rehydration would be required afterwards.

Reply:
Yep, that's the intent - it'd probably require some sort of jenkins slave cleanup in the cleanup script though... but that's for a follow on.